yellow-paper/docs/state/index.md

```diff
@@ -4,12 +4,43 @@ sidebar_position: 10
 
-Merkle trees are either 
-- [append-only](./tree_impls.md#append-only-merkle-trees), for data where we only require inclusion proofs or 
-- [indexed](./tree_impls.md#indexed-merkle-trees) for storing data that requires proofs of non-membership.
+When dealing with private data, only the hash of the data is stored in the leaf in our data tree and we must setup a derivation mechanism that ensures nullifiers can be computed deterministically from the pre-image (the data that was hashed). This way, no-one can tell what data is stored in the leaf (unless they already know it), and therefore won't be able to derive the nullifier and tell if the leaf is active or deleted. 
+
+Convincing someone that a piece of data is active can then be done by proving its membership in the data tree, and that it is not deleted by proving its non-membership in the nullifier tree. This ability to efficiently prove non-membership is one of the extra requirements we have for some parts of our state. To support the requirements most efficiently, we use two families of Merkle trees:
+- The [Append-only Merkle tree](./tree_impls.md#append-only-merkle-trees), which supports efficient membership proofs,
+- The [Indexed Merkle tree](./tree_impls.md#indexed-merkle-trees), which supports efficient membership and non-membership proofs but increase cost of adding leafs.
+
+### Private State Access
+
+Whenever a user is to read of use data, they must then convince the "rollup" that the their data is active. As mentioned above, they must prove that the data is in the data tree (membership proof) and that it is still active (non-membership proof). However, there are nuances to this approach! 
+
+One important aspect is *when* state can be accessed. In most blockchains, state is always accessed at the head of the chain and changes are only made by the sequencer as new blocks are added. However, since private execution relies on proofs generated by the user this would be very impractical - one users transaction would invalidate everyone elses. 
+
+While proving inclusion in the data tree can be done using historical state, the non-membership in the nullifier tree cannot. Membership can be proven since we are using a append-only tree, so anything that was there in the past, must still be in the tree now. However, this don't work for the non-membership proof as it would only prove that the data was active at the time the proof was generated, not that it is still active! Which would allow a user to create multiple transactions spending the same data and then send them all at once - double-spending!
```

Comment:
``` suggestion

While proving inclusion in the data tree can be done using historical state, the non-membership proof in the nullifier tree cannot. 

Membership can be proven using historical state because we are using an append-only tree, so anything that was there in the past must still be in the append-only tree now. 

However, this doesn't work for the non-membership proof, as it can only prove that the data was active at the time the proof was generated, not that it is still active today! This would allow a user to create multiple transactions spending the same data and then send those transactions all at once, creating a double spend. 

```